ClickHouse.Client/Formats/BinaryStreamReader.cs

```diff
@@ -120,7 +131,19 @@ public object Read(IPv4Type pv4Type)
 
         public object Read(EnumType enumType) => enumType.Lookup(reader.ReadSByte());
 
-        public object Read(NestedType tupleType) => throw new NotSupportedException();
+        public object Read(NestedType tupleType)
+        {
+            // => throw new NotSupportedException();
+            var count = tupleType.UnderlyingTypes.Length;
+            var contents = new object[count];
+            for (var i = 0; i < count; i++)
+            {
+                var value = Read(tupleType.UnderlyingTypes[i]);
+                contents[i] = ClearDBNull(value);
+            }
+            //return (1, "vx1" );
+            return tupleType.MakeTuple(contents);
```

Comment:
Tuple is read fine here

Reply:
Shouldn't this return an array of `Tuple`s?

Reply:
The value is represented as a single `Tuple`

Reply:
`Nested` is an [array of tuples](https://clickhouse.tech/docs/en/sql-reference/data-types/nested-data-structures/nested/), not a single tuple - how does this work when there are multiple values inside?

Reply:
> `Nested` is an [array of tuples](https://clickhouse.tech/docs/en/sql-reference/data-types/nested-data-structures/nested/), not a single tuple - how does this work when there are multiple values inside?

That is correct, because the Array type is wrapped around the nested type it's still read that way. But an atomic nested value needs to be read as a single tuple.

You can already see this in the unit test; ShouldInsertIntoSubNestedTableViaBulk but i've also extended the unit test; ShouldInsertIntoSubSubNestedTableViaBulk so you can see it works with sub nested values too; https://github.com/olafbuitelaar/ClickHouse.Client/blob/master/ClickHouse.Client.Tests/NestedTableTests.cs#L142